src/build_vehicle_gui.cpp

```diff
@@ -1610,6 +1607,9 @@ struct BuildVehicleWindow : Window {
 
 		this->SetWidgetsDisabledState(this->sel_engine == INVALID_ENGINE, WID_BV_SHOW_HIDE, WID_BV_BUILD, WID_BV_RENAME, WIDGET_LIST_END);
 
+		/* Disable renaming engines in network games if you are not the server. */
+		this->SetWidgetDisabledState(WID_BV_RENAME, _networking && !_network_server);
```

Comment:
Doesn't this now enable it for servers while the line above it wants it disabled?